sympy/solvers/tests/test_solvers.py

```diff
@@ -1845,3 +1845,25 @@ def test_inf():
     assert solve(1 - oo*x) == []
     assert solve(oo*x, x) == []
     assert solve(oo*x - oo, x) == []
+
+
+def test_issue_12448():
+    from time import time
+    e, s = S('''
+        ([a(0, 0)*x(0) + a(0, 1)*x(1) + a(0, 2)*x(2) + b(0),
+        a(1, 0)*x(0) + a(1, 1)*x(1) + a(1, 2)*x(2) + b(1),
+        a(2, 0)*x(0) + a(2, 1)*x(1) + a(2, 2)*x(2) + b(2)],
+        [x(0), x(1), x(2)])''')
+    t = time()
+    ans = solve(e, s)
+    tf = time() - t
+    e, s = S('''
+    ([a_00*x_0 + a_01*x_1 + a_02*x_2 + b_0, a_10*x_0 +
+    a_11*x_1 + a_12*x_2 + b_1, a_20*x_0 + a_21*x_1 +
+    a_22*x_2 + b_2], (x_0, x_1, x_2))''')
+    t = time()
+    ans = solve(e, s)
+    ts = time() - t
+    # the two solutions should be found in nearly the same amount
+    # of time.
+    assert min(tf, ts)/max(tf, ts) > 0.9
```

Comment:
Interesting result: the function form, rather than the symbol form, is faster by almost a factor of 3 whether it is run first or second.  (Relative times were 2.3 vs 6.5, a factor of 2.8.)